File: 2017/day1/day1a.py
sums = 0

with open('input') as f:
    line = f.readline()
    for i in range(len(line) - 2):
        if line[i] == line[i + 1]:
            sums += int(line[i])

    if line[-1] == line[0]:
        sums += int(line[0])

print(sums)

File: 2017/day1/day1a.hs
import System.IO
import Data.Char

addIfEqual :: Int -> Int -> Int
addIfEqual a b
    | a == b = a
    | otherwise = 0

moveHeadToTail :: [Int] -> [Int]
moveHeadToTail (x:xs) = xs ++ [x]

pairs :: [Int] -> [Int]
pairs x = zipWith addIfEqual x (moveHeadToTail x)

solve :: String -> Int
solve x = sum $ pairs $ map digitToInt x

main = do
    withFile "input" ReadMode (\handle -> do  
        line <- hGetLine handle
        putStr $ show $ solve line)
